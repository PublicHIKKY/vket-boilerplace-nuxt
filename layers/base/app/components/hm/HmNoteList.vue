<template>
  <ul class="hm-note-list">
    <li
      v-for="(text, index) in list"
      :key="`note-${index}`"
      class="item"
    >
      {{ text }}
    </li>
  </ul>
</template>

<script lang="ts" setup>
defineProps<{
  list: string[]
}>()
</script>

<style lang="scss" scoped>
@use '#base/app/assets/styles/variables' as v;

.hm-note-list {
  font-size: 14px;
  line-height: 1.3;

  > .item {
    margin-bottom: 4px;

    &::before {
      content: 'â€»';
      margin-right: 0.3em;
    }
  }
}
</style>
