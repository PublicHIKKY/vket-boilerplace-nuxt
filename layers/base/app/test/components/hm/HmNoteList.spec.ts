import { mount } from '@vue/test-utils'
import HmNoteList from '#base/app/components/hm/HmNoteList.vue'

test('ref component', () => {
  expect(HmNoteList).toBeTruthy()
})

test('mount component', () => {
  const wrapper = mount(HmNoteList, {
    props: {
      list: [],
    },
  })
  expect(wrapper.getCurrentComponent()).toBeTruthy()
  expect(wrapper.html()).toMatchSnapshot()
})

describe('props', () => {
  test(':list', () => {
    const wrapper = mount(HmNoteList, {
      props: {
        list: ['list1', 'list2'],
      },
    })
    expect(wrapper.findAll('.item')[0]?.text()).toBe('list1')
    expect(wrapper.findAll('.item')[1]?.text()).toBe('list2')
  })
})
