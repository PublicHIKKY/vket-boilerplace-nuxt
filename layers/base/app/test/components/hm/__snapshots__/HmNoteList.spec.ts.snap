// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mount component 1`] = `"<ul data-v-91fb3b34="" class="hm-note-list"></ul>"`;
